flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/partitioner/GlobalPartitioner.java

```diff
@@ -40,6 +41,16 @@ public int selectChannel(SerializationDelegate<StreamRecord<T>> record) {
 		return this;
 	}
 
+	@Override
+	public ChannelStateRescaler getUpstreamChannelStateRescaler() {
+		return ChannelStateRescaler.FIRST_CHANNEL;
+	}
+
+	@Override
+	public ChannelStateRescaler getDownstreamChannelStateRescaler() {
+		return ChannelStateRescaler.ROUND_ROBIN;
```

Comment:
I'm probably misunderstanding it, but it seems it should be the opposite:
upstream: `ROUND_ROBING` (or both FIRST)
downstream: `FIRST_CHANNEL`

Reply:
Yes changed that (see comment below). On upstream it should be almost always round robin (broadcasting partitioner is the only exception where we drop extra state).